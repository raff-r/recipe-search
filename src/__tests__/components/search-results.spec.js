import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json";
import MockResponse from '../mock-response.json';

import SearchResults from '../../components/search-results';

describe('SearchResults Component', () => {

    const mockStore = configureStore();
    const initialState = {
        results: [{
            "recipe" : {
                "uri" : "http://www.edamam.com/ontologies/edamam.owl#recipe_f1c853a77986214680bbdd424883499a",
                "label" : "Herbes de Provence Rotisserie Chickens",
                "image" : "https://www.edamam.com/web-img/18d/18dcf05995cb40e8ce4c077972341d7a.jpg",
                "source" : "Bon Appetit",
                "url" : "http://www.bonappetit.com/recipe/herbes-de-provence-rotisserie-chickens",
                "shareAs" : "http://www.edamam.com/recipe/herbes-de-provence-rotisserie-chickens-f1c853a77986214680bbdd424883499a/chicken",
                "yield" : 6.0,
                "dietLabels" : [ "Low-Carb" ],
                "healthLabels" : [ "Sugar-Conscious", "Peanut-Free", "Tree-Nut-Free", "Alcohol-Free" ],
                "cautions" : [ ],
                "ingredientLines" : [ "2 tablespoons (1/4 stick) butter, room temperature", "2 tablespoons dried herbes de provence*", "1 tablespoon coarse kosher salt", "2 (3 1/2-pound) chickens" ],
                "ingredients" : [ {
                    "text" : "2 tablespoons (1/4 stick) butter, room temperature",
                    "weight" : 28.399999618530273
                }]
            }
        }]
    };
    let component, store;

    beforeEach(() => {
        store = mockStore(initialState);
        component = mount( <SearchResults store={store}/> );
    });

    it('should match the snapshot', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

});