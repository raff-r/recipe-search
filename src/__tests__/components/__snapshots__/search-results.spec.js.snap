// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Component should match the snapshot 1`] = `
<Connect(SearchResults)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchResults
    dispatch={[Function]}
    results={
      Array [
        Object {
          "recipe": Object {
            "cautions": Array [],
            "dietLabels": Array [
              "Low-Carb",
            ],
            "healthLabels": Array [
              "Sugar-Conscious",
              "Peanut-Free",
              "Tree-Nut-Free",
              "Alcohol-Free",
            ],
            "image": "https://www.edamam.com/web-img/18d/18dcf05995cb40e8ce4c077972341d7a.jpg",
            "ingredientLines": Array [
              "2 tablespoons (1/4 stick) butter, room temperature",
              "2 tablespoons dried herbes de provence*",
              "1 tablespoon coarse kosher salt",
              "2 (3 1/2-pound) chickens",
            ],
            "ingredients": Array [
              Object {
                "text": "2 tablespoons (1/4 stick) butter, room temperature",
                "weight": 28.399999618530273,
              },
            ],
            "label": "Herbes de Provence Rotisserie Chickens",
            "shareAs": "http://www.edamam.com/recipe/herbes-de-provence-rotisserie-chickens-f1c853a77986214680bbdd424883499a/chicken",
            "source": "Bon Appetit",
            "uri": "http://www.edamam.com/ontologies/edamam.owl#recipe_f1c853a77986214680bbdd424883499a",
            "url": "http://www.bonappetit.com/recipe/herbes-de-provence-rotisserie-chickens",
            "yield": 6,
          },
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <h1>
        Search Results
      </h1>
      <SearchResult
        result={
          Object {
            "cautions": Array [],
            "dietLabels": Array [
              "Low-Carb",
            ],
            "healthLabels": Array [
              "Sugar-Conscious",
              "Peanut-Free",
              "Tree-Nut-Free",
              "Alcohol-Free",
            ],
            "image": "https://www.edamam.com/web-img/18d/18dcf05995cb40e8ce4c077972341d7a.jpg",
            "ingredientLines": Array [
              "2 tablespoons (1/4 stick) butter, room temperature",
              "2 tablespoons dried herbes de provence*",
              "1 tablespoon coarse kosher salt",
              "2 (3 1/2-pound) chickens",
            ],
            "ingredients": Array [
              Object {
                "text": "2 tablespoons (1/4 stick) butter, room temperature",
                "weight": 28.399999618530273,
              },
            ],
            "label": "Herbes de Provence Rotisserie Chickens",
            "shareAs": "http://www.edamam.com/recipe/herbes-de-provence-rotisserie-chickens-f1c853a77986214680bbdd424883499a/chicken",
            "source": "Bon Appetit",
            "uri": "http://www.edamam.com/ontologies/edamam.owl#recipe_f1c853a77986214680bbdd424883499a",
            "url": "http://www.bonappetit.com/recipe/herbes-de-provence-rotisserie-chickens",
            "yield": 6,
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="col-xs-12"
          >
            <h3>
              Herbes de Provence Rotisserie Chickens
            </h3>
          </div>
          <div
            className="col-xs-6 col-md-4"
          >
            <img
              alt="Herbes de Provence Rotisserie Chickens"
              className="img-responsive"
              src="https://www.edamam.com/web-img/18d/18dcf05995cb40e8ce4c077972341d7a.jpg"
            />
          </div>
          <div
            className="col-xs-6 col-md-3"
          >
            <h4>
              Diet labels
            </h4>
            <ul
              className="list-inline"
            >
              <li>
                <span
                  className="label label-info"
                >
                  Low-Carb
                </span>
              </li>
            </ul>
            <h4>
              Health labels
            </h4>
            <ul
              className="list-inline"
            >
              <li>
                <span
                  className="label label-info"
                >
                  Sugar-Conscious
                </span>
              </li>
              <li>
                <span
                  className="label label-info"
                >
                  Peanut-Free
                </span>
              </li>
              <li>
                <span
                  className="label label-info"
                >
                  Tree-Nut-Free
                </span>
              </li>
              <li>
                <span
                  className="label label-info"
                >
                  Alcohol-Free
                </span>
              </li>
            </ul>
            <h4>
              Calories
            </h4>
            <p>
              N/A
            </p>
          </div>
          <div
            className="col-xs-12 col-md-5"
          >
            <h4>
              Ingredients
            </h4>
            <ul>
              <li>
                2 tablespoons (1/4 stick) butter, room temperature
                 - 
                28.40
                g
              </li>
            </ul>
          </div>
        </div>
      </SearchResult>
    </div>
  </SearchResults>
</Connect(SearchResults)>
`;
